### ISSUE TITLE
feat: add ridge regressor SFD

### ISSUE THREAD

**Issue by mikkokotila** [Vaquum/Limen#825]
## Slice ID

ridge-sfd

## Slice Title

feat: add ridge regressor SFD

## Primary Task Type

capability

## Routing and Dependencies

> **Significance.** Routing is the minimum information needed to place this
> slice in the dependency graph. Parent and dependencies must be existing
> issue numbers; unlocks may be forward-looking. Prose description of the
> relationship is not accepted — only issue references. This section is not
> a planning document; it is a graph edge list.

Parent issue: #297
Depends on: none
Unlocks: none

## Capability

> **Significance.** A slice delivers exactly one capability. The capability
> must be expressible as a set of assertions that become mechanically true
> at merge. This is the answer to *what changes in the system that an
> external observer can verify by running a command?* Every line below
> must be evaluable as a shell command, SQL query, Python expression, or
> HTTP request that returns a definite result. One assertion per line.
>
> Accepted:
> - `python -c "from limen.inference import Trainer; print(Trainer.__name__)"` returns `Trainer`.
> - `python -m tests.run` returns exit code `0`.
> - `ruff check .` returns exit code `0`.
>
> Rejected:
> - *Freeze the current-truth read contract.*
> - *Strict CQRS separation.*
> - *Terminal proof obligations are complete.*

Assertions (one per line, each executable as-is):
- `python -m pytest tests/test_ridge_regressor.py -k matches_sklearn` exits 0.
- `python -m pytest tests/test_ridge_regressor.py -k evaluation_contract` exits 0.
- `python -m pytest tests/test_ridge_regressor.py -k yaml_experiment_reconstructs` exits 0.

## Objective, Baseline, and Scope

> **Significance.** Objective, baseline, and scope define the smallest
> useful change. The objective states the outcome. Baseline is repo or
> runtime state today, expressed as commands and outputs a reader can run
> against `main`. Scope names the exact surfaces this slice may touch.
>
> Accepted:
> - `git show HEAD:pyproject.toml | grep -c 'name = "limen"'` returns `1`.
> - `test -f limen/cli/main.py && echo present` returns `present`.
>
> Rejected:
> - *The system has accumulated technical debt.*
> - *Several paths are confusing.*

Objective: expose ordinary scikit-learn ridge regression through the existing Limen SFD interfaces.
Baseline at `38f89b60`:
- `test -f limen/sfd/reference_architecture/ridge_regressor.py` exits 1.
- `test -f limen/sfd/foundational_sfd/ridge_regressor.py` exits 1.
- `test -f limen/yaml/templates/ridge_regressor.yaml` exits 1.
Scope: the paths in Surfaces only.

## Design

> **Significance.** Design is the end-to-end shape of the solution. Name
> the concrete files, commands, interfaces, and data flow that make the
> capability true. Do not restate the objective.

Design:
- `RidgeRegressor(ReferenceModel)` delegates each fit and prediction to `sklearn.linear_model.Ridge`; no custom solver, cross-round caches, or benchmark paths.
- `ridge_regressor(data, alpha=1.0, fit_intercept=True, copy_X=True, max_iter=None, tol=1e-4, solver='auto', positive=False, random_state=42)` returns standard continuous metrics, `_preds`, `_model`, and existing directional inline metrics.
- `foundational_sfd/ridge_regressor.py` supplies `params()` and a normal `MLManifest`; the matching YAML template uses existing indicators, lag features, train-fitted RobustScaler, and NextReturnTarget.
- Existing UEL, Trainer, Sensor, backtest, and parameter plumbing remain unchanged. Numerical backend errors and warnings retain their sklearn meaning.
- Tests reuse the checked-in real-market parquet fixture, with no downloaded or generated data.

## Runtime and Observability Scope

> **Significance.** Runtime and observability scope states whether this
> slice changes execution behavior, environment, telemetry, logs, metrics,
> alerts, or operational proof. If there is no runtime touch, say so.

Runtime touch: the new ridge SFD only; scikit-learn is already a runtime dependency.
Observability touch: existing regression, directional confusion, and optional backtest outputs; no new metric or log schema.
Operational proof: `python -m tests.run tests/test_ridge_regressor.py`.

## Proof Coverage

> **Significance.** Proof coverage names the commands that must return the
> stated result after merge. These commands are the proof obligation of the
> slice. Each one maps to a test or CI gate in the Proof section below.
> Vague targets like "the system is faster" or "the contract is hardened"
> are rejected — state the measurable predicate.
>
> Accepted:
> - `python -m tests.run` returns exit code `0`.
> - `ruff check .` returns exit code `0`.
> - `python -m coverage report --fail-under=92` returns exit code `0`.

Commands with expected exit code 0:
- `python -m tests.run tests/test_ridge_regressor.py`
- `python -m tests.run`
- `ruff check limen governance tests`
- `python governance/check_module_budgets.py`
- CI `.github/workflows/pr_checks_typing.yml` runs strict Pyright and the protected-base typing ratchet with required artifacts.
- Required PR checks, including package install, coverage, docs, and packaging.

## Surfaces

> **Significance.** This is the mechanical scope contract. List every file
> path that the PR delivering this slice will modify, add, or remove.
> Files outside these three lists must not change. A scope-check CI step
> compares the PR's changed-file set against this list and fails if they
> diverge. The scope model is: "out of scope is every file not listed
> here." Glob patterns are permitted; prose is not.

Modified:
- `limen/sfd/reference_architecture/__init__.py`
- `limen/sfd/foundational_sfd/__init__.py`
- `limen/sfd/reference_architecture/README.md`
- `limen/sfd/foundational_sfd/README.md`
- `limen/sfd/README.md`
- `docs/Reference-Architecture.md`
- `docs/Built-In-SFDs.md`
- `tests/test_foundational_sfd.py`
- `.github/module_budgets.json`
- `pyproject.toml`
- `CITATION.cff`
- `CHANGELOG.md`
- `limen/yaml/templates/*.yaml`
- `docs/End-to-End-Workflow.md`
- `docs/Experiment-Manifest.md`

Added:
- `limen/sfd/reference_architecture/ridge_regressor.py`
- `limen/sfd/foundational_sfd/ridge_regressor.py`
- `limen/yaml/templates/ridge_regressor.yaml`
- `tests/test_ridge_regressor.py`

Removed: none

## Signature Diff

> **Significance.** For every surface in the list above that has a public
> signature — a function, a class, a SQL view or table, an HTTP route, a
> CLI entry point, a JSON schema — give the `Before` and `After` forms as
> code blocks. These signatures are the contract future PRs cannot
> silently regress. Prose like "rename the function" or "make it
> fail-loud" is rejected — the only accepted content under `Before` and
> `After` is a literal code fragment in the language of the surface.

### Reference architecture exports
Before:
```python
__all__ = [...]
```
After (additional exports):
```python
__all__ = [..., 'RidgeRegressor', 'ridge_regressor']
class RidgeRegressor(ReferenceModel):
    def train(self, data: dict[str, object], **params: object) -> 'RidgeRegressor': ...
    def predict(self, data: dict[str, object]) -> dict[str, _Predictions]: ...
    def evaluate(self, data: dict[str, object], inline_metrics: bool = True) -> dict[str, object]: ...
def ridge_regressor(data: dict[str, object], alpha: float = 1.0,
                    fit_intercept: bool = True, copy_X: bool = True,
                    max_iter: int | None = None, tol: float = 1e-4,
                    solver: str = 'auto', positive: bool = False,
                    random_state: int | None = 42) -> dict[str, object]: ...
```
### Foundational SFD export
Before:
```python
__all__ = [...]
```
After (additional export and new module):
```python
__all__ = [..., 'ridge_regressor']
def params() -> dict[str, list[object]]: ...
def manifest() -> MLManifest: ...
```
### Bundled YAML template
Before:
```yaml
# No ridge_regressor template.
```
After:
```yaml
reference_architecture: limen.sfd.reference_architecture.ridge_regressor.ridge_regressor
```

## Proof

> **Significance.** Every assertion in the *Capability*
> section above has exactly one test. This section is the mapping. Each
> row names the MVC assertion (quote or shorten it), the test file path,
> and the test name within that file. The test must exist at merge and
> must pass in CI. Prose like "terminal proof obligation," "claim class,"
> or "proof freshness requirement" is rejected — the row's middle column
> is a path that resolves to a running test, or the row is wrong.

| MVC assertion | test file | test name |
|---|---|---|
| sklearn parity | tests/test_ridge_regressor.py | test_ridge_matches_sklearn |
| standard evaluation outputs | tests/test_ridge_regressor.py | test_ridge_evaluation_contract |
| YAML experiment and reconstruction | tests/test_ridge_regressor.py | test_ridge_yaml_experiment_reconstructs |

## Guardrails

> **Significance.** A slice is not complete if its capability can be
> silently removed by the next PR. Name the CI workflow and job that
> will fail a future PR that regresses this MVC. If no such gate exists
> today, creating it is part of this slice. The answer below is the
> mechanical block on tomorrow's PR undoing today's work.
>
> Accepted:
> - Workflow `.github/workflows/pr_checks_tests.yml`, job
>   `pr_checks_tests`, which runs `python -m coverage run -m tests.run`
>   and fails the PR on non-zero exit.
>
> Rejected:
> - *A guardrail will be added to the backfill protocol.*
> - *Invariants documented in the slice spec.*
> - *The proof fixture will be extended.*

Workflow file: `.github/workflows/pr_checks_tests.yml`
Job name: `pr_checks_tests`
Specific failing command: `python -m coverage run -m tests.run` (nonzero test status fails the job).

## Step Register

> **Significance.** Step Register is the ordered execution backlog for
> this slice. Each step must name a concrete edit, command, or GitHub
> action. It is not a narrative plan.

Steps:
1. Add sklearn-backed class/function, foundational SFD, YAML template, exports, and focused tests.
2. Update catalogs, add budgets for the two new modules, and bump 5.10.3 to 5.11.0 with synchronized version fields.
3. Commit and open the PR; run local checks and inspect the published diff.
4. Record test results, resolve relevant review findings, and request operator review.

## Out of Scope

> **Significance.** The slice PR must not touch the paths listed here.
> A scope-check CI step compares the PR diff against this list and fails
> if any out-of-scope path appears in the diff. This is the mechanical
> complement to the *Surfaces* section above. "Out of scope" here is
> not aspirational — it is a CI-enforced block.

Paths that must not change:
- `limen/experiment/**`
- `limen/inference/**`
- `limen/backtest/**`
- `limen/metrics/**`
- `limen/data/**`
- `limen/features/**`
- `limen/scalers/**`
- `rust/**`
- `scripts/**`
- `requirements/**`
- `.github/workflows/**`
- `.github/typing_budget.json`
- `.github/fail_loud_budget.json`

## Risks and Open Questions

> **Significance.** Short. Only risks that change the decision tree for
> this slice belong here. A risk that is resolved by doing more
> engineering work is engineering work, not a risk, and goes in the
> *Proof Coverage* section instead.

Risks: backend solver and floating-point behavior are not promised to match an external implementation's pivot cutoff or failures.
Open questions: none; external comparison harnesses and performance measurements are not part of this PR.

## Done Means

> **Significance.** Done Means closes on verifiable evidence, not on a
> self-assessed claim. Check each item only when the linked evidence
> exists. Every checkbox above must be checked or marked
> `OVERRULED: <reason>` before the delivering PR can merge (slice gate
> rule 10). The evidence fields below are written automatically by
> `slice_closeout_guard` when a merged PR closes this issue; a close
> with no merged PR and no evidence is reverted.

- [x] Capability complete
- [x] Proof complete
- [x] Guardrails complete
- [x] Required gates green
- [x] Required docs updated

Evidence: PR #826, commit `c1b572a1`; local full suite 1348 passed / 1 existing opt-in skip, focused ridge suite 33 passed, both new Python modules 100% statement coverage. Implementation gates pass on `c1b572a1`: tests/coverage/runtime run [35711991414](https://github.com/Vaquum/Limen/actions/runs/35711991414), typing [35711991452](https://github.com/Vaquum/Limen/actions/runs/35711991452), packaging and Python 3.10–3.13 installs [35711991481](https://github.com/Vaquum/Limen/actions/runs/35711991481), docs [35711991413](https://github.com/Vaquum/Limen/actions/runs/35711991413), lint [35712109618](https://github.com/Vaquum/Limen/actions/runs/35712109618). Slice-gate rerun follows this evidence update; human approval and merge are not claimed.

Merge SHA: 44eff3b244c80ce8473845bd60b074b3682fe441
Merged PR number: #826
Required CI runs (workflow name : run id):
- PR Checks CodeQL (python) : 106694539400
- pr_checks_cc : 106694541245
- pr_checks_lint : 106694539320
- pr_checks_ruleset : 106694539259
- pr_checks_slice : 106696149428
- pr_checks_fail_loud : 106694539301
- pr_checks_typing : 106694539282
- pr_checks_version : 106694539629
- pr_checks_honesty : 106694539337
- PR Checks Tests : 106694539043
- PR Checks Coverage : 106695313997
- PR Checks Runtime : 106695314042
- PR Checks Docs Site : 106694539152
- PR Checks Packaging : 106694539079
- PR Checks Package Install (3.10) : 106694539413
- PR Checks Package Install (3.11) : 106694539295
- PR Checks Package Install (3.12) : 106694539352
- PR Checks Package Install (3.13) : 106694539401

## Author Checks

- [x] The Significance blockquote at the top of every section above remains in this issue unchanged.
- [x] Every MVC assertion is executable (shell command, SQL query, Python expression, or HTTP request) and has a definite expected output.
- [x] Every MVC assertion has exactly one row in the Tests table pointing at a concrete test file and test name.
- [x] Every Surfaces entry with a public signature has a corresponding Signature Diff entry with literal Before and After code blocks.
- [x] Guardrails cites a CI workflow file path, a job name, and a specific failing command — not a promise to add one later.
- [x] Objective, Baseline, and Scope entries and Proof Coverage entries contain commands, not prose.
